e2e/rook-test-framework/clients/rook_client.go

```diff
@@ -0,0 +1,102 @@
+package clients
+
+import (
+	"errors"
+	"github.com/rook/rook/e2e/rook-test-framework/contracts"
+	"github.com/rook/rook/e2e/rook-test-framework/enums"
+	"github.com/rook/rook/e2e/rook-test-framework/transport"
+)
+
+var (
+	STATUS_CMD  = []string{"rook", "status"}
+	VERSION_CMD = []string{"rook", "version"}
+	NODE_CMD    = []string{"rook", "node"}
+)
+
+type RookClient struct {
+	platform        enums.RookPlatformType
+	transportClient contracts.ITransportClient
```

Comment:
I definitely like what you guys are doing with interfaces so that multiple platforms can be supported and the tests can interact with them in an standardized way via the interfaces.